tofhir-server/src/main/scala/io/tofhir/server/service/terminology/codesystem/CodeSystemRepository.scala

```diff
@@ -194,7 +219,55 @@ class CodeSystemRepository(terminologySystemFolderPath: String) extends ICodeSys
    * @param codeSystemId  id of the code system
    * @return Source of the csv file
    */
-  override def getCodeSystemContent(terminologyId: String, codeSystemId: String): Future[Source[ByteString, Any]] = {
+  override def getCodeSystemContent(terminologyId: String, codeSystemId: String, pageNumber: Int, pageSize: Int): Future[(Source[ByteString, Any], Long)] = {
+      // check if code system id exists in json file
+    val localTerminologyFile = FileUtils.getPath(getTerminologySystemsJsonPath(terminologySystemFolderPath)).toFile
+    val localTerminology = FileOperations.readJsonContent[TerminologySystem](localTerminologyFile)
+
+    localTerminology.find(_.id == terminologyId) match {
```

Comment:
Can you please add a descriptive comment here on what you are checking? By associating it with an example file structure maybe. What is inside a terminology file etc.